src/__experimental__/components/date/date.component.js

```diff
@@ -110,15 +105,20 @@ class BaseDateInput extends React.Component {
   openDatePicker = () => {
     this.isBlurBlocked = true;
     document.addEventListener('click', this.closeDatePicker);
-    this.updateSelectedDate(this.props.value);
+    this.updateSelectedDate(this.props.value || DateHelper.formatValue(this.state.visibleValue));
     this.setState({ isDatePickerOpen: true });
   };
 
   reformatVisibleDate = () => {
     const { visibleValue } = this.state;
-
     if (DateHelper.isValidDate(visibleValue)) {
-      this.setState({ visibleValue: formatDateToCurrentLocale(visibleValue) });
+      this.setState({
+        visibleValue: formatDateToCurrentLocale(visibleValue),
+        lastValidEventValues: {
+          formattedValue: formatDateToCurrentLocale(visibleValue),
+          rawValue: DateHelper.formatValue(visibleValue)
+        }
```

Comment:
There are 3 places in this file where this identical object structure is created:
```js
{
  visibleValue: formatDateToCurrentLocale(visibleValue),
  lastValidEventValues: {
    formattedValue: formatDateToCurrentLocale(visibleValue),
    rawValue: DateHelper.formatValue(visibleValue)
  }
}
```
It'd be good to DRY that up, by creating a helper function, which takes a `visibleValue` parameter, and creates and returns that object :point_up:

Reply:
Yep agree I'll reformat the code 👍

Reply:
Thanks for refactoring with that new `updateValidEventValues()` method :thumbsup: